ui/app/src/components/DatasourceList/DatasourceDrawer.tsx

```diff
@@ -21,16 +21,19 @@ import { DatasourceEditorForm } from './DatasourceEditorForm';
 
 export type ActionStr = 'Create' | 'Update';
 
-interface DatasourceDrawerProps {
-  datasource: Datasource;
+type DispatchWithPromise<A> = (value: A) => Promise<void>;
```

Comment:
Move to a utils file?
I remember seeing this same type declared in the variable PR :p